opencog/query/PatternMatchEngine.cc

```diff
@@ -1307,9 +1307,12 @@ bool PatternMatchEngine::clause_accept(const Handle& clause_root,
 	}
 	if (not match) return false;
 
-	clause_grounding[clause_root] = hg;
-	logmsg("---------------------\nclause:", clause_root);
-	logmsg("ground:", hg);
+	if (not is_evaluatable(clause_root))
+	{
+		clause_grounding[clause_root] = hg;
+		logmsg("---------------------\nclause:", clause_root);
+		logmsg("ground:", hg);
+	}
```

Comment:
Not a bad solution, but i'm wondering if this is hiding another bug.  Evaluatable clauses are not supposed to be grounded, they're supposed to be evaluated, and so I'm somewhat surprised that they are even going through this particular block of code. 

I don't have time to dig into this right now; I think I need to open to issues to track this so I don't forget.